src/manager/agent.manager.service.ts

```diff
@@ -128,6 +133,47 @@ export class AgentManagerService {
         return res.data.invitation;
     }
 
+    private async pingConnectionWithRetry(agentId: string, adminPort: string, adminApiKey: string, durationMS: number = 10000) : Promise<any> {
+        Logger.info(`pingConnectionWithRetry`);
+        const compute = (l , r) => {
+            let result = l.getTime() - r.getTime();
+            if (result <= 0) {
+                result = (result + 1000) % 1000;
+            }
+            return result;
+        };
+
+        const http = new ProtocolHttpService(new HttpService());
```

Comment:
Maybe add a TODO (or a mental note) something like we do in createConnection: 
TODO move to it's own class and pass in the http object

Reply:
done